### DISCUSSION TITLE
Considerations for a 1.0.0 RC3 update

### DISCUSSION THREAD

**Discussion by pagbabian-splunk** [ocsf/ocsf-schema#530] [Ideas]
# RC 3 Items for consideration
The following items have been proposed over the last few months that merit consideration for a RC3 update with the intention of becoming the 1.0.0 GA release.  Some have been addressed or partially addressed but please review and comment.  Time is of the essence, as we don't want to drag things on too long, but important changes should be considered for GA release especially if they will entail post GA breaking changes, or are of fundamental importance to adopters.
## Analytic object (Proposal #497) - Merged PR #526
- Is related_analytics a list of objects or names?  Recursive or not?
- Fix vulnerability (title to name, needs an array of CVEs). (new Proposal) as well as related_vulnerabilities
- Can we create a single solution for all the related_xx attributes?  Built like related_event
## Security Finding (Proposal #490 #491 #492) - Merged PR #526
- Impact_score, along with normal sibling impact
## Remove Confidence from Base (Proposal  #514)
Can improve Security Finding, pair with sibling
## Resource, Managed Entity (Issue #396, Proposal #499, Proposal #518, PR #528)
- Group_name -> group (object)
- Remove, file, device (don’t add user), (don’t add policy)
- Cloud_partition, account_uid, region?  Should be part of cloud profile or object.
- Src_url?  Should this be a string for URI, is it the same as uid, or is it a true URL?
- Remove System Specific enum value
- Win_resource profile should be registered and will replace the enum list (type_id)
## Windows Resource
- Remove the object, use a profile, and add the enum list to the profile; apply to Resource; Keep Windows Resource Activity, fix up the resource object, and the activity_id (Roumen)
## Windows Profile - apply to Resource, and activities (Issue #428)
- Goes with above (Windows Resource)
- Add windows event code
- Otherwise just the Host profile is in the class
## Cloud object / cloud_partition
- Move cloud_partition from Resource to the cloud object or cloud profile
## HTTP Activity (Discussion #485)
- Activity_id need to reflect the layer 7 protocol (Ania)
- http_method needs an enum of methods
- http_headers becomes headers (KV pairs)
- (don’t include the removal of Access Activity in #485)
## Access Activity
- Needs work, not ready for RC3, but we can’t remove as it is used
- Needs a Resource at minimum (don’t assume HTTP)
## Audit category name
- Consider AAA again
- Consider moving to System Activity
- It’s about controlling access – Consider Access Control / User Activity (accessing resources)
## hash attribute (Issue #515)
- merged from PR
## Network Profile (?) (Issue #429)
- Closed for now – no use cases identified
## Cloud related attributes (Issue #404)
- E.g. we don’t want cloud specific attrs in Device
- resolve org_unit: it is in the cloud object and the device object
## Add an organization attribute (?). (new Proposal, not the same as org_unit – e.g. LDAP)
- Relates to the SaaS profile proposal #519 (Odi)
## Account object (Discussion #495) (Ania)
- Accounts are different than users
- We have a number of account_xx attributes
- Accounts are different than organizations
## Correlation Details (Issue #342) - replace correlation_uid
- Closed and referred to Finding Details object for who correlated what events
## Extension Versioning (Proposal #477)
- Merged from PR
## URL Class update (Proposal #506)
- Create an Issue
## SaaS Profile (Proposal #519)
- Looks good, needs more discussion
- Too late for RC3?
## Managed Entity
- Combine with Resource, update the class
## Is Configuration/Inventory ready for RC3 / 1.0?
- Possibly remove category
- The classes could be in System Activity and Access Control (Audit)
- Possibly create Device Activity in System Activity and add attribute or profile (CIS Benchmark Result)
- Device Inventory Info class has no new attributes and only the Log activity_id
## Add customer_uid to Metadata,
- eg. For multi-tenant use case (new Proposal) (Roumen)
- refer to proposal #519 (Odi)
## Add names to enums in addition to captions (Poll #313)
- Big change, but voted on last October 2022
- FYI: enums can be text, or names, along with captions; do we need all three?
- It would force us to fill in all the existing enums, many won’t need short names

**Comment by jasonbreimer** [comment]
As the owner of this Category I've not heard any discussion on removing Configuration/Inventory. Please bring this discussion to channel: inventory-state.

I would not be in favor or removing an entire Category. This is not activity data or Event data. There is no meaningful timestamp for this data and as stateful data doesn't match "activity" data types. We spent many months working through the category and I think it warrants some deep discussion before making such a significant change.

**Reply by JasonKeirstead** [reply]
I agree with Jason on this.

The object is already actively in use by more than one party. To pull it at this stage when people are actually using it and there is not a strong criticism or consensus of a change, does not make a lot of sense to me.

**Comment by floydtree** [comment]
Agree, I am in favor keeping the Configuration/Inventory category as is for rc3.

**Comment by pagbabian-splunk** [comment]
The category has been converted to Discovery in RC3.
